StudentNotes/StudentNotes.Infrastructure/Repositories/NotesRepository.cs

```diff
@@ -0,0 +1,100 @@
+﻿using StudentNotes.Application.IRepositories;
+using StudentNotes.Core.Entities.Notes;
+using Microsoft.EntityFrameworkCore;
+using StudentNotes.Infrastructure.ApplicationContext;
+
+namespace StudentNotes.Infrastructure.Repositories
+{
+    public class NotesRepository : INotesRepository
+    {
+        private DbSet<NoteBase> _table;
+
+        private EFContext _db;
+
+        public async Task AddAsync(NoteBase item)
+        {
+            await this._table.AddAsync(item);
+            this.SaveAsync();
+        }
+
+        public async Task DeleteAsync(NoteBase item)
+        {
+            this._table.Remove(item);
+            this.SaveAsync();
+        }
+
+        public async Task<NoteBase> GetNoteAsync(int id)
+        {
+            return await this._table.FirstOrDefaultAsync<NoteBase>(entity => entity.Id == id);
```

Comment:
Use the same logic as for the Methods where you get collections of notes, so add include and so on